SLA Algorithm/Sample_SLA_Algorithm_1.go

```diff
@@ -0,0 +1,90 @@
+package main
+
+import (
+	"sort"
+)
+
+func calculateValue(values []float64, input_range float64) bool {
+	if input_range <= values[1] && values[0] <= input_range {
+		return true
+	} else {
+		return false
+	}
+}
+
+func main() {
+	var user_input = map[int]string{ // Input from the user for each of the SLI
+		1: "node_load",
+	}
+	var priority = make([]int, 0, len(user_input)) // Storing all the priority in each a list and will sort it,as GO iterates over map randomly
+	for name := range user_input {
+		priority = append(priority, name)
+	}
+	sort.Ints(priority) //The sorting according to priority of each of the SLI
+
+	var budget float64 = 10                       // Budget for the application critical metrics and will reduce if the any of critical metrics goes down
+	var input_node_load = []float64{4.0, 8.0}     // User defined node load of each of the requests (SLO)
+	var input_response_time = []float64{3.0, 9.0} // User defined response time of each of the requests (SLO)
+	var input_SLA float64 = 99.00                 // SLA given by the user
+	var list_of_uptimes = []float64{}             // The list fo all the previous uptime
+	var downtime_delta = []float64{}              // The list fo all the difference between two uptime
+	var uptime_state bool = true                  // Storing the state of uptime for calculating delta
+	var new_uptime float64 = 1000                 // The new uptime value pulled from prometheus in seconds
+	var total_time float64 = 10000                // Total time for which the application is running from prometheus
+	var downtime = total_time - new_uptime        // Downtime for the application
+	var n int = len(list_of_uptimes) - 1          // The length of all the elements in previous uptime
+	var aggregated_node_load float64 = 7          // The aggregated value from prometheus for node load
+	var aggregated_response_time float64 = 7      // The aggregated value from prometheus for response time
+
+	if total_time != new_uptime && len(list_of_uptimes) == 0 && uptime_state == true { // When downtime happens for the first time
+		uptime_state = false
+		list_of_uptimes = append(list_of_uptimes, new_uptime)
+	} else if list_of_uptimes[n] != new_uptime { // When the uptime begins after a downtime and storing the value of Delta
```

Comment:
I think, this algorithm could be simplified, with a simple function - getStatusOfApp(AppName) 
- Get the current status of app, if it's down, note the time we checked and do a delta between when the app changed status and when we checked last time, if it's down, we already broke the SLA(unless we're checking for part of apps). With this you don't have to maintain, another slice of uptime and downtime_delta.